lexicons/app/bsky/embed/record.json

```diff
@@ -10,24 +10,36 @@
         "record": {"type": "ref", "ref": "com.atproto.repo.strongRef"}
       }
     },
-    "presented": {
+    "view": {
       "type": "object",
       "required":  ["record"],
       "properties": {
-        "record": {"type": "union", "refs": ["#presentedRecord", "#presentedNotFound"]}
+        "value": {"type": "union", "refs": ["#viewRecord", "#viewNotFound"]}
       }
     },
-    "presentedRecord": {
+    "viewRecord": {
       "type": "object",
-      "required":  ["uri", "cid", "author", "record"],
+      "required":  ["uri", "cid", "author", "record", "indexedAt"],
       "properties": {
         "uri": {"type": "string", "format": "at-uri"},
         "cid": {"type": "string", "format": "cid"},
         "author": {"type": "ref", "ref": "app.bsky.actor.defs#withInfo"},
-        "record": {"type": "unknown"}
+        "record": {"type": "unknown"},
+        "embeds": {
```

Comment:
is this an array or just a union?

Reply:
A single record only has one embed right?

Reply:
A post record has at most one embed, but if we want this to remain generic for any record type, I figured there could be multiple embeds.
> If there are multiple, it would be up to the consumer to distinguish them by inspecting `embed.value.record`.

Reply:
Hmmm i think im inclined to make it only one embed 🤔 

If you want multiple embeds then you define a new type for that in the union. Similar to how we handle images which is actually up to 4 embeds but represented as only a single embed

This let's us set constraints on how many objects of what type are embedded instead of an unconstrained array

Thoughts?

Reply:
I guess that also only works if this union is only for schemas that _we_ control. because another developer could take an alternate approach and have some record contain an array of embeds. But I suppose then we could just define a new single embed type that points to that array of embeds.  🤔

Reply:
Idk im not fully convinced either way. But I think when we originally had this convo about embeds we chose not to make it an array because then we can set constraints on each embed type. Everything is still expressible, it might just be one semantic layer deeper

Reply:
I see what you're getting at 👍 I do think I still lean towards the array but I'm also definitely not married to it.  Part of my view on it is treating embeds the same way we might treat any other type: if record embed views needed to present all blobs that appear in the record, for example, I imagine we might end-up with an array of blobs here.  That would be necessary to cover cases like profile records which have multiple blobs, i.e. both an avatar and a background banner.  I was imagining a similar case where a record could conceivably contain two or more separate embed fields, since there's nothing restricting that.

Here's a case to illustrate where my head was at!  Consider a "question and answer" record, where an actor can present a post that poses a question next to a post that answers that question.  It might be modeled like this, with an embed for the question post and another for the answer post:
```json
{
  "commentary": {"type": "string"},
  "questionEmbed": {"type": "ref", "ref": "app.bsky.embed.record"},
  "answerEmbed": {"type": "ref", "ref": "app.bsky.embed.record"}
}
```

At the same time I'd be totally cool saying that we're just not going to cater to that for record embeds, that there's a convention that you should avoid creating records with multiple embed fields 👍 We also do always have the option to go back on the idea of this being a generic record embed, and instead making it specifically a post embed.

Reply:
The argument would be that the presentation of that possibility would be
```json
"questionAndAnswerEmbedView": {
  "questionEmbed": {"type": "ref", "ref": "app.bsky.embed.record"},
  "answerEmbed": {"type": "ref", "ref": "app.bsky.embed.record"}
}
```

And similarly any array of heterogenous embeds can be encapsulated in a lex type that boils it down to a single value that's expressive about what exactly is in the array.

With that being said, it starts to sound a bit overly complex/galaxy brained as a way of encoding what's essentially a list of supplemental "view" data

Reply:
I think I'd feel more strongly if this was primary data. But since it's supplemental data & the actual structure is imposed by the record, I'm game to leave this field relatively unstructured & just return an array of heterogenous embeds.

Reply:
Cool cool 👍 it's just a little tricky to deal with view information for any records in a unified way.  In the case of a complex record like the question-and-answer one, I would expect a consumer to discover embeds in the original record data.  And then fish the view information for those embeds out of this array.

I imagine same might go for blobs if we introduce those here: you discover a blob in the original record, and then fish the view of that blob (e.g. a url to a resized image) out of some collection/array of blob views.

Reply:
yup yup that checks out. im game with just the array here. thx for the back and forth 🙏